src/Doctrine/Instantiator/Instantiator.php

```diff
@@ -94,7 +94,7 @@ private function buildFactory(string $className) : callable
 
         $serializedString = sprintf(
             '%s:%d:"%s":0:{}',
-            self::SERIALIZATION_FORMAT_AVOID_UNSERIALIZER,
+            is_subclass_of($className, \Serializable::class) ? self::SERIALIZATION_FORMAT_USE_UNSERIALIZER : self::SERIALIZATION_FORMAT_AVOID_UNSERIALIZER,
```

Comment:
Hmm, can you run CS locally? I think it will complain about some formatting and imports.

Otherwise, patch looks pristine!